uktrade/data-hub-frontend
# New service questions for interactions

## Problem
Data for interaction services is changing, so some services for interactions will now have additional questions added and some services are being split into two options which can also have additional questions to follow. Adding the follow on questions are straight forward as we can get this from the backend however at this stage we cannot split the service options into two in the backend.

## Solution
We have decided to update the service data in the frontend so that the services that need to be split will be split into two by a colon being set between the primary and secondary service option.

### For example

Split with a colon `Providing Export Advice & Information : Advice & information`

**Account Management
Providing Export Advice & Information : Advice & information
Providing Export Advice & Information : Something else**

#### Would result in this structure

**Account Management
Providing Export Advice & Information 
— Advice & information
Providing Export Advice & Information
— Something else**

This is just a temporary solution and the backend will soon be updated and services will be split in the back end properly so there is a tech debt clean up to happen afterwards.


## Test instructions

1. Go to add an interaction
2. Select services and choose various options

You should then get a variety of single select dropdown options with follow on questions along with double select dropdown options with follow on questions.  
 
## Screenshots
### Before

![Screenshot 2019-06-28 at 10 00 43](https://user-images.githubusercontent.com/10154302/60333132-2f55c000-9990-11e9-82de-e6f16ad545e3.png)


### After 

![Screenshot 2019-06-28 at 10 00 18](https://user-images.githubusercontent.com/10154302/60333142-34b30a80-9990-11e9-8a88-d659e5c097a8.png)


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied. 
https://github.com/uktrade/data-hub-frontend/blob/develop/docs/Code%20review%20guidelines.md"
